fastlane_bot/events/exchanges/solidly_v2.py

```diff
@@ -57,17 +57,70 @@ async def _get_tkn0_B(contract: Contract) -> str:
-    "xfai_v0"       : {"decimals":  4, "factory_abi": XFAI_V0_FACTORY_ABI       , "pool_abi": XFAI_V0_POOL_ABI   , "get_fee": _get_fee_7, "get_tkn0": _get_tkn0_B, "get_tkn1": _get_tkn1_B},
+    "velocimeter_v2": {"decimals":  4, "factory_abi": VELOCIMETER_V2_FACTORY_ABI, "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_1, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "equalizer_v2"  : {"decimals":  4, "factory_abi": SCALE_V2_FACTORY_ABI      , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_2, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "aerodrome_v2"  : {"decimals":  4, "factory_abi": SOLIDLY_V2_FACTORY_ABI    , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_3, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_2, "get_pool_args": _get_pool_args_1},
+    "velodrome_v2"  : {"decimals":  4, "factory_abi": SOLIDLY_V2_FACTORY_ABI    , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_3, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_2, "get_pool_args": _get_pool_args_1},
+    "scale_v2"      : {"decimals": 18, "factory_abi": SCALE_V2_FACTORY_ABI      , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_2, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "cleopatra_v2"  : {"decimals":  4, "factory_abi": CLEOPATRA_V2_FACTORY_ABI  , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_4, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "stratum_v2"    : {"decimals":  4, "factory_abi": VELOCIMETER_V2_FACTORY_ABI, "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_1, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "lynex_v2"      : {"decimals":  4, "factory_abi": LYNEX_V2_FACTORY_ABI      , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_5, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "nile_v2"       : {"decimals":  4, "factory_abi": NILE_V2_FACTORY_ABI       , "pool_abi": SOLIDLY_V2_POOL_ABI, "get_fee": _get_fee_6, "get_tkn0": _get_tkn0_A, "get_tkn1": _get_tkn1_A, "get_pool_function": _get_pool_function_1, "get_pool_args": _get_pool_args_1},
+    "xfai_v0"       : {"decimals":  4, "factory_abi": XFAI_V0_FACTORY_ABI       , "pool_abi": XFAI_V0_POOL_ABI   , "get_fee": _get_fee_7, "get_tkn0": _get_tkn0_B, "get_tkn1": _get_tkn1_B, "get_pool_function": _get_pool_function_2, "get_pool_args": _get_pool_args_2},
 }
```

Comment:
This looks like a complete overkill.

You can probably just use the following 3 functions:
```
def _get_pool_1(factory_contract, tkn0, tkn1, stable):
    return factory_contract.functions.getPair(tkn0, tkn1, stable)

def _get_pool_2(factory_contract, tkn0, tkn1, stable):
    return factory_contract.functions.getPool(tkn0, tkn1, stable)

def _get_pool_3(factory_contract, tkn0, tkn1, stable):
    return factory_contract.functions.getPool(tkn0)
```
Then combine the `get_pool_function` attributes and the `get_pool_args` attributes into a single attribute which will be set to one of these 3 functions on each entry in the table above.

Reply:
Not relevant any more